src/serde_arrow_type.erl

```diff
@@ -0,0 +1,58 @@
+%%%-----------------------------------------------------------------------------
+%% @doc The `serde_arrow_type' module provides types in order to represent Apache
+%% Arrow specific primitive logical datatypes, as well as Erlang datatypes which
+%% `serde_arrow' supports serializing and deserializing.
+%% @end
+%%%-----------------------------------------------------------------------------
+-module(serde_arrow_type).
+-export_type([
+    arrow_type/0,
+    arrow_bool/0,
+    arrow_int/0,
+    arrow_uint/0,
+    arrow_float/0,
+    erlang_type/0
+]).
+
+-type arrow_type() ::
+    arrow_bool()
+    | arrow_int()
+    | arrow_uint()
+    | arrow_float().
+%% Any primitive logical type in Apache Arrow that is supported by `serde_arrow'.
+
+-type arrow_bool() ::
+    arrow_true
+    | arrow_false.
+%% Apache Arrow Boolean. One of `True' or `False'
+
+-type arrow_int() ::
+    {int, 8}
+    | {int, 16}
+    | {int, 32}
+    | {int, 64}.
```

Comment:
```suggestion
    {s, 8}
    | {s, 16}
    | {s, 32}
    | {s, 64}.
```

At least so we align with Nx? WDYT?

Reply:
At the same time, Nx does not have booleans or lists, etc. So maybe this alignment would be silly. :)

Reply:
No, I think this a valid suggestion. I felt that `int` was a little verbose.